third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -385,13 +385,22 @@ impl<'env> Generator<'env> {
                 // appear where references are processed.
                 let rhs_temp = self.gen_arg(rhs, false);
                 let lhs_temp = self.gen_auto_ref_arg(lhs, ReferenceKind::Mutable);
-                if !self.temp_type(lhs_temp).is_mutable_reference() {
+                let lhs_type = self.get_node_type(lhs.node_id());
+
+                // For the case: `fun f(p: &mut S) { *(p :&S) =... }, we need to check whether `lhs_type` is immutable ref
```

Comment:
This comment is just confusing.  What is `lhs_type` in the move code you show?  Is it `&S` or `S`?

Reply:
revised the comment